lib/IRGen/GenDecl.cpp

```diff
@@ -1105,8 +1105,11 @@ void IRGenerator::emitGlobalTopLevel(llvm::StringSet<> *linkerDirectives) {
     IGM->emitSILDifferentiabilityWitness(&dw);
   }
 
-  // Emit code coverage mapping data.
-  PrimaryIGM->emitCoverageMapping();
+  // Emit code coverage mapping data for all modules
+  for (auto Iter : *this) {
+    IRGenModule *IGM = Iter.second;
+    IGM->emitCoverageMapping();
+  }
```

Comment:
I'm not sure how order-sensitive all this code is so I left this here in its own loop but I'm happy to merge this into another one if that is safe!

Reply:
This looks fine to me, the steps here seem fairly decoupled.